lens-follower-tracker/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+mod abis;
+mod pb;
+mod helpers;
+
+use pb::lens::events::{FollowNftDeployed, FollowEvents, FollowNftTransferred};
+use substreams::errors::Error;
+use substreams::{Hex};
+use substreams_ethereum::{pb::eth::v2 as eth, Event as EventTrait};
+use helpers::utils::format_with_0x;
+
+
+// Lens LPP contract 
+// const LENS_LPP_CONTRACT : &str = "db46d1dc155634fbc732f92e853b10b288ad5a1d";
+const ZERO_ADDRESS : &str = "0x0000000000000000000000000000000000000000";
+const LENS_PROXY_ACTION_ADDRESS : &str = "0xdd3f6c22ecc68007cc9f76da18984995da4b7b82";
+const FOLLOW : &str = "follow";
+const UNFOLLOW : &str = "unfollow";
+
+substreams_ethereum::init!();
+
+/// Extracts lens events from the contract
+#[substreams::handlers::map]
+fn map_lens_follower_events(contract_address: String, blk: eth::Block) -> Result<FollowEvents, Error> {
+    let mut follow_nft_deployed_events = vec![];
+    let mut follow_nft_transferred_events: Vec<FollowNftTransferred> = vec![];
+
+    for log in blk.logs() {
+        let addres : String = Hex(log.address().clone()).to_string();
+        if !contract_address.contains(&addres) {
+           continue;
+        }
+        // for FollowNftDeployed event
+        if let Some(event) = abis::lens_events::events::FollowNftDeployed::match_and_decode(log) {
+            let follow_nft_deployed_event : FollowNftDeployed = FollowNftDeployed {
+                follow_profile_id: event.profile_id.to_string(),
+                follow_token_address: helpers::utils::format_with_0x(Hex(event.follow_nft).to_string())
+            };
+            follow_nft_deployed_events.push(follow_nft_deployed_event)
```

Comment:
Execution can continue from this point. No need to decode other events.

Reply:
done